### PR TITLE
add byte serializer library

### PR DESCRIPTION
**Pull request by bogdan34**

### CODE REVIEW

**Review by axelsheva — APPROVED** [review]

File: src/byte_serializer/models/array.ts [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+
+    getBodyLength(value: Array<any>): number {
+        let size = 0;
+        value.forEach(item => {
+            size += this.typeElement.getLength(item);
+        });
+        return size;
+    }
+
+    writeArrayElementTypeId(buffer: Buffer, elementTypeId: number, offset: number): number {
```
Comment (Verbalist) [inline src/byte_serializer/models/array.ts]: private for all private methods

File: src/byte_serializer/models/uint64.ts [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+    private _stringNullGenerator(count: number): string {
+        let str = '';
+        for (let i = 0; i < count; i++) {
+            str = '0' + str;
+        }
+        return str;
+    }
+
+    read(buffer, offset) {
+        const MAX = 32;
```
Comment (Verbalist) [inline src/byte_serializer/models/uint64.ts]: const in function

**Review by Verbalist — APPROVED** [review]

File: src/byte_serializer/models/map/index.ts [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+        this.val = val;
+        this.typeElement = new SimpleArray(new MapObject(key, val));
+    }
+
+    getLength(data: Map<any, any>) {
+        const arr = [...data].map(([key, val]) => ({ key, val }));
+
+        const size = this.length.type + this.length.type + this.length.type + this.typeElement.getLength(arr);
+
+        return size;
```
Comment (macaqueee) [inline src/byte_serializer/models/map/index.ts:23]: is it redundant? space for logs? :D

File: src/byte_serializer/models/map/mapObject.ts [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+
+    constructor(key: ModelType, value: ModelType) {
+        super();
+        this.localSchema = new LocalSchema(key, value);
+    }
+
+    getLength(object): number {
+        let bufferSize = 0;
+
+        Object.keys(this.localSchema).forEach((key) => {
```
Comment (macaqueee) [inline src/byte_serializer/models/map/mapObject.ts]: how about to use `reduce` instead of `forEach` here?

**Pull request merged**